instrumentation/cassandra/cassandra-4.4/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_4/CassandraAttributesExtractor.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.cassandra.v4_4;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
+import com.datastax.oss.driver.api.core.cql.Statement;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import javax.annotation.Nullable;
+
+final class CassandraAttributesExtractor
+    implements AttributesExtractor<CassandraRequest, ExecutionInfo> {
+
+  @Override
+  public void onStart(
+      AttributesBuilder attributes, Context parentContext, CassandraRequest request) {}
+
+  @Override
+  public void onEnd(
+      AttributesBuilder attributes,
+      Context context,
+      CassandraRequest request,
+      @Nullable ExecutionInfo executionInfo,
+      @Nullable Throwable error) {
+    if (executionInfo == null) {
+      return;
+    }
+
+    Node coordinator = executionInfo.getCoordinator();
+    if (coordinator != null) {
+      if (coordinator.getDatacenter() != null) {
+        attributes.put(SemanticAttributes.DB_CASSANDRA_COORDINATOR_DC, coordinator.getDatacenter());
+      }
+      if (coordinator.getHostId() != null) {
+        attributes.put(
+            SemanticAttributes.DB_CASSANDRA_COORDINATOR_ID, coordinator.getHostId().toString());
+      }
+    }
+    attributes.put(
+        SemanticAttributes.DB_CASSANDRA_SPECULATIVE_EXECUTION_COUNT,
+        executionInfo.getSpeculativeExecutionCount());
+
+    // TODO can this be rewritten using newer API?
+    @SuppressWarnings("deprecation")
+    Statement<?> statement = executionInfo.getStatement();
```

Comment:
@SimoneGiusso can you look at this to see if there's a better way to do this in 4.4+

Reply:
They [made `ExecutionInfo` compatible with any `Request` type](https://github.com/datastax/java-driver/pull/1367/files), and the current deprecated method just do a cast. We can do the [same](https://github.com/datastax/java-driver/blob/355844408325df5045fc5b10d99737f24003042d/core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java#L324).